### ISSUE TITLE
Background Worker written in Go

### ISSUE THREAD

**Issue by fabriziomello** [microo8/plgo#2]
Hi,

We did some experiments, look at: https://github.com/prest/bgworker

**Comment by microo8** [comment]
Yes I've seen that and add a star to that project :) But didn't have time/energy/some other excuse ...
I leave this open as a reminder to do it. Or if anyone can do it I will pull it right away

**Comment by omani** [comment]
any news to this? being able to use goroutines would be absolutely awesome.

**Comment by microo8** [comment]
I've been testing this. It works fine, also have ideas how to integrate it to plgo tool. But I've got a problem, that goroutines are not starting.
Is someone familiar with it? I must somehow enable threading in the background worker?

**Comment by fabriziomello** [comment]
@microo8 would be nice if you share some code sample

**Comment by microo8** [comment]
The code that I experimented with is in the [test/bgw](https://github.com/microo8/plgo/tree/master/test/bgw) directory

Wanted to isolate the `get_got _sigterm` and `wait_latch` to something similar as the [`time.Tick`](https://golang.org/pkg/time/#Tick) method. It needs to run a separate goroutine to wait for the latch. (Also need to add `close` on that channel)

Also tried to run a goroutine, that doesn't use the log (maybe the log isn't accessible from another thread), but does something else (like touch a new file). But it doesn't run.
